lib/ome/files/MetadataTools.cpp

```diff
@@ -548,7 +548,8 @@ namespace ome
           try
             {
               OMEXMLMetadata& omexml(dynamic_cast<OMEXMLMetadata&>(store));
-              addMetadataOnly(omexml, s);
+              if (omexml.getTiffDataCount(s) == 0)
```

Comment:
For complete correctness, I think you are spot on.  With this change, I'm copying the Java MetadataTools verbatim.  The other oddity is that it (and `addMetadataOnly`) are using `OMEXMLMetadata` rather than just `Metadata`.